### PR TITLE
Refactor kubelet collector test

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Rewrite `TestNoMetricsCollected` with [testutils.CollectAndCompare](https://github.com/kubernetes/kubernetes/pull/83699).

We need to remove Prometheus reference to move [metrics stability](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/20190404-kubernetes-control-plane-metrics-stability.md) into Beta.

By the way, that might be a problem in the origin `TestNoMetricsCollected` implementation.
Because we can't send messages to a channel (`ch := make(chan prometheus.Metric)`) without a receiver. 
In other words, it will cause `fatal error: all goroutines are asleep - deadlock!` when the test failing.

**Which issue(s) this PR fixes**:
Ref https://github.com/kubernetes/enhancements/issues/1238

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Provided the test suite continues to pass, I don't see any downsides to this change - I think it moves us in the direction we want to go.

Thanks!

**Comment by RainbowMango** [comment]
Solved conflict with https://github.com/kubernetes/kubernetes/pull/83704.

**Comment by RainbowMango** [comment]
Kindly ping @dashpole @sjenning 

/priority important-soon

**Review by logicalhan — APPROVED** [review]
/lgtm
